Source/UI/NewUI/PaymentForm/ViewController/PaymentViewController.swift

```diff
@@ -114,6 +119,22 @@ extension PaymentViewController {
         }
     }
 
+    private func setupCardNumberViewClosure() {
+        viewModel.updateCardNumberView = { [weak self] in
+            DispatchQueue.main.async {
+                guard let self = self else { return }
+                self.updateCardNumber()
+            }
+        }
+    }
+
+    private func updateCardNumber(){
+        guard let style = viewModel.paymentFormStyle?.cardNumber else { return }
+        cardNumberView.iconView.update(with: "icon-visa".image(forClass: CardListCell.self))
+        cardNumberView.iconView.isHidden = false
+        cardNumberView.update(style: style)
```

Comment:
Will that image be dynamically adjusted later? 

Also, for the UIStackView visibility change, I'd wrap that inside a `UIView.animate` block, its always nice to get that free smooth animation from the UIStackView 😉.

Reply:
Wonder whether the icon view update should also be internal to the `update`. Feels like it would better enable the view to protect its subviews. Also, there should be a path for `isHidden` = false, but that may come with the validation?